crates/sui-adapter/src/execution_engine.rs

```diff
@@ -425,20 +402,33 @@ fn advance_epoch<S: BackingPackageStore + ParentSync + ChildObjectResolver>(
         );
         temporary_store.reset();
         let function = ADVANCE_EPOCH_SAFE_MODE_FUNCTION_NAME.to_owned();
-        adapter::execute::<execution_mode::Normal, _, _>(
+        let safe_mode_pt = {
+            let mut builder = ProgrammableTransactionBuilder::new();
+            let res = builder.move_call(
+                (*module_id.address()).into(),
+                module_id.name().to_owned(),
+                function,
+                vec![],
+                vec![
+                    system_object_arg,
+                    CallArg::Pure(bcs::to_bytes(&change_epoch.epoch).unwrap()),
+                    CallArg::Pure(bcs::to_bytes(&change_epoch.protocol_version).unwrap()),
+                ],
+            );
+            assert_invariant!(
+                res.is_ok(),
+                "Unable to generate advance_epoch_safe_mode transaction!"
+            );
+            builder.finish()
+        };
+        programmable_transactions::execution::execute::<_, _, execution_mode::Normal>(
```

Comment:
Heads up to @lxfind and @amnn, this is the new way to make system transaction calls :)

Reply:
n00b q: we are only calling one Move function here. Why do we need to use PT?

Reply:
That's all there is anymore. The normal `adapter` is dying in this PR